lib/providers/first_party.yml

```diff
@@ -89,3 +89,14 @@ audio-api:
 documentcloud:
   endpoint: "https://www.documentcloud.org/api/oembed.json"
   pattern_list: ["#\\Ahttps?://www\\.documentcloud\\.org/documents?/.+#i"]
+
+
+# Facebook oEmbed documentation: https://developers.facebook.com/docs/plugins/oembed-endpoints/
+
+facebook-video:
+  endpoint: "https://www.facebook.com/plugins/video/oembed.json"
+  pattern_list: ["#https://www\\.facebook\\.com/.+/videos/.+/#i", "#https://www\\.facebook\\.com/.+/videos/.+/#i", "#https://www\\.facebook\\.com/video\\.php\\?id=.+#i", "#https://www\\.facebook\\.com/video.php\\?v=.+#i"]
+
+facebook-post:
+  endpoint: "https://www.facebook.com/plugins/post/oembed.json"
+  pattern_list: ["#https://www.facebook.com/.+/posts/.+#i", "#https://www.facebook.com/.+/posts/.+#i", "#https://www.facebook.com/.+/activity/.+#i", "#https://www.facebook.com/photo.php\\?fbid=.+#i", "#https://www.facebook.com/photos/.+#i", "#https://www.facebook.com/permalink.php\\?story_fbid=.+#i", "#https://www.facebook.com/media/set\\?set=.+#i", "#https://www.facebook.com/questions/.+#i", "#https://www.facebook.com/notes/.+/{note-url}/.+#i"]
```

Comment:
Should the dots in the hostname part be escaped since they're a regexp wildcard character (like on line 98)?

I think the `{note-url}` at the end of this line is part of the URL template on Facebook's oEmbed documentation page and not part of the literal URL that this gem will process.

Reply:
oooooops. I'll clean that up.